{"tags": ["mysql", "django"], "answers": [{"score": 0, "last_activity_date": 1570181783, "answer_id": 58233709, "question_id": 58233475, "body": "<p>you have to use field names</p>\n\n<pre><code>queryset = User_info.objects.all().prefetch_related(\"P_key\")\nqueryset = User_info.objects.all().select_related(\"P_key\")\n</code></pre>\n"}], "is_answered": true, "answer_count": 1, "score": 0, "last_activity_date": 1570181783, "creation_date": 1570180914, "question_id": 58233475, "title": "Trouble with joining tables together", "body": "<p>I'm trying to join tables by using Django queryset, but for some reason it keeps throwing an error.</p>\n\n<p>Tables are structured as below.</p>\n\n<pre><code>class Platform(models.Model):\n    P_key = models.AutoField(primary_key=True)\n    P_url = models.CharField(max_length=100)\n    P_userkey = models.IntegerField(default=0)\n    P_name = models.CharField(max_length=10)\n    objects = models.Manager()\n\nclass User_info(models.Model):\n    U_key = models.AutoField(primary_key=True)\n    U_name = models.CharField(max_length=20)\n    U_img = models.CharField(max_length=100)\n    U_info = models.CharField(max_length=100)\n    U_sudate = models.CharField(max_length=20)\n    P_key = models.ForeignKey(Platform, on_delete=models.CASCADE)\n    objects = models.Manager()\n</code></pre>\n\n<p>This is the code written to join two tables together.</p>\n\n<pre><code>queryset = User_info.objects.all().prefetch_related(\"Platform\")\nqueryset = User_info.objects.all().select_related(\"Platform\")\nqueryset = Platform.objects.all().select_related(\"User_info\")\n\n</code></pre>\n\n<p>And the following is the error:</p>\n\n<blockquote>\n  <p>django.core.exceptions.FieldError: Invalid field name(s) given in\n  select_related: 'Platform'. Choices are: P_key.</p>\n</blockquote>\n\n<p>I've tried a number of query sets but I wasn't able to get far.</p>\n"}